### PR TITLE
1438561: Do not use D-Bus for facts collection

### PR DESCRIPTION
**Pull request by kahowell**
Instead, we use the same underlying service, which is also exposed via
D-Bus.

This involved refactoring the aggregation into a separate service -
`AllFactsCollector`, and then using that to implement `AllFacts`.

I also moved the default configuration of the collectors into their
classes.

Also, I removed now outdated comments and logging that suggested we are
using D-Bus for facts collection.

### CODE REVIEW

File: src/rhsmlib/facts/collector.py [hunk 1]
```diff
@@ -102,7 +108,7 @@ def get_all(self):
 
 
 class StaticFactsCollector(FactsCollector):
-    def __init__(self, static_facts, **kwargs):
+    def __init__(self, static_facts=DEFAULT_STATIC_FACTS, **kwargs):
```
Comment (awood) [inline src/rhsmlib/facts/collector.py]: This worries me a little bit.  I worry that someone creates a dictionary of static facts and sends it in and that will mean that "system.certificate_version" will be undefined (unless the person remembers to define it in their dictionary).  Not having "system.certificate_version" set will mean the client will get version 1 entitlement certificates which is definitely not what we want.

How about instead:

```python
def __init__(self, static_facts=None, **kwargs):
  if static_facts is None:
    static_facts = {}
  static_facts.set_default("system.certificate_version, "3.2")
  ...
```

Or something similar.  Basically just something to insure that "system.certificate_version" is defined even if the dictionary `__init__` receives doesn't have it.

**Comment by kahowell** [comment]
9c5bd51..f27dafb: use `setdefault` for certificate version.

**Comment by awood** [comment]
Looks good to me, but let's see what @barnabycourt has to say about the root/non-root stuff.

**Comment by barnabycourt** [comment]
@awood @kahowell I would love to make sub-man not require root but outside of a general plan to get to that point the benefits to the ability to debug by not routing through d-bus are substantial.

**Comment by kahowell** [comment]
@barnabycourt ack. I put an item on the backlog to come up with a general plan. I'll go ahead and merge this work. @awood fyi

**Pull request merged**